samples/unixdomainsockets/src/main/java/okhttp3/unixdomainsockets/ClientAndServer.java

```diff
@@ -34,7 +35,9 @@ public void run() throws Exception {
     socketFile.delete(); // Clean up from previous runs.
 
     MockWebServer server = new MockWebServer();
-    server.setServerSocketFactory(new UnixDomainServerSocketFactory(socketFile));
+    RealOkioServerSocketFactory serverSocketFactory =
+      new RealOkioServerSocketFactory(new UnixDomainServerSocketFactory(socketFile));
```

Comment:
possibly worth a TODO? Feels like this new API supporting other socket types would be a win?